cpp/src/io/parquet/writer_impl.cu

```diff
@@ -2184,6 +2184,9 @@ writer::impl::impl(std::vector<std::unique_ptr<data_sink>> sinks,
   if (options.get_metadata()) {
     _table_meta = std::make_unique<table_input_metadata>(*options.get_metadata());
   }
+  if (_write_v2_headers and _compression == Compression::ZSTD) {
+    CUDF_FAIL("V2 page headers cannot be used with ZSTD compression");
```

Comment:
we need the same check for the chunked case as well

Reply:
Oops. Or should the test (and the lines above it) go in `init_state()` instead? (And maybe change the name of `init_state`).

Reply:
Looks like I'm late with the answer:)
Side note: the current code in `init_state` should not be there (breaks the strong exception guarantee)